src/coinjoin/client.h

```diff
@@ -72,10 +70,50 @@ class CPendingDsaRequest
     }
 };
 
+struct CJClientManager {
+    CJClientManager(CConnman& connman, CTxMemPool& mempool, const CMasternodeSync& mn_sync,
+                    const std::unique_ptr<CCoinJoinClientQueueManager>& queueman)
+        : m_connman(connman), m_mempool(mempool), m_mn_sync(mn_sync), m_queueman(queueman) {}
+    ~CJClientManager() {
+        for (auto& pair : m_imap) {
+            pair.second.reset();
+        }
+    }
+
+    void Add(CWallet& wallet);
+    void DoMaintenance(CBlockPolicyEstimator& fee_estimator);
+
+    void Remove(const std::string& name) {
+        auto it = m_imap.find(name);
+        if (it != m_imap.end()) { m_imap.erase(it); }
+    }
+
+    CCoinJoinClientManager* Get(const CWallet& wallet) const {
+        auto it = m_imap.find(wallet.GetName());
+        return (it != m_imap.end()) ? it->second.get() : nullptr;
+    }
+
+    std::map<const std::string, std::unique_ptr<CCoinJoinClientManager>>& raw() { return m_imap; }
```

Comment:
I don't love the name `raw`, please make it something like `GetWalletCJManMap` or something like that

Reply:
I think `raw` is an acceptable function name as it's a simple description of what it does, giving you underlying access to the member that `CJClientManager` is a convenience wrapper for (and also an opportunity to add provide guardrails instead of leaving it exposed in the global scope).

Though I have marked the returned type and the function itself as `const` in the most recent push.

Reply:
It's really not clear that `CJClientManager` is just a wrapper of the wallet name -> CCoinJoinClientManager; In fact I'd say having two classes one named CCoinJoinClientManager and the other CJClientManager is confusing

Reply:
Understandable. I could rename it to `CJClientsManager` but I doubt a single-letter difference would help much in differentiation. Currently, the way to tell them apart is the difference in naming convention but down the road, perhaps `CCoinJoinClientManager` could be renamed to `CJClientEntity` or something in the spirit of it being instantiated as an element in a collection (in this case, a map).

Though, any such refactoring would fall outside the scope of this PR.